runtime/cudaq/dynamics/cudm_state.cpp

```diff
@@ -0,0 +1,252 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include <cmath>
+#include <cudaq/cudm_state.h>
+#include <iostream>
+#include <numeric>
+#include <sstream>
+#include <stdexcept>
+
+namespace cudaq {
+
+cudm_state::cudm_state(std::vector<std::complex<double>> rawData)
+    : rawData_(rawData), state_(nullptr), handle_(nullptr),
+      hilbertSpaceDims_() {
+  HANDLE_CUDM_ERROR(cudensitymatCreate(&handle_));
```

Comment:
We should only have a single `cudensitymatHandle_t` throughout the code (i.e., a single `cudensitymatCreate` and `cudensitymatDestroy`).

For example, we already create a handle in `test_cudm_time_stepper.cpp` (`SetUp`); thus creating another handle here is problematic.

I'd suggest passing the handle here, e.g. 

```
cudm_state::cudm_state(cudensitymatHandle_t handle,
                       const std::vector<std::complex<double>> &rawData)
    : rawData_(rawData), state_(nullptr), handle_(handle), hilbertSpaceDims_() 
 ```